fe1-web/src/features/evoting/network/messages/CastVote.ts

```diff
@@ -90,4 +91,35 @@ export class CastVote implements MessageData {
       lao: new Hash(obj.lao),
     });
   }
+
+  /**
+   * Generates a vote id as described in
+   * https://github.com/dedis/popstellar/blob/master/protocol/query/method/message/data/dataCastVote.json
+   * HashLen('Vote', election_id, question_id, (vote_index(es)|write_in)), concatenate vote indexes - must use delimiter
+   * @param election The election for which this vote id is generated
+   * @param questionIndex The index of the question in the given election, this vote is cast for
+   * @param selectionOptions The selected answers for the given question in the given eeleciton
+   */
+  public static computeVoteId(
+    election: Election,
+    questionIndex: number,
+    selectionOptions: Set<number>,
+  ): Hash {
+    return Hash.fromStringArray(
+      EventTags.VOTE,
+      election.id.toString(),
+      election.questions[questionIndex].id,
+      // Important: A standardized order is required, otherwise the hash cannot be verified
+      // Even more important: A standardized delimiter has to be used to disambiguate [1,0] from [10]
+      // See https://github.com/dedis/popstellar/issues/843 for details
+      // TODO: Update after discussion in #843 is finished
```

Comment:
TODO - just a reminder 😄

Reply:
Yep - But this will probably have to wait at least until the meeting today